File: Java/Euler004.java
/**
 * Euler Problem 4:
 * 
 * A palindromic number reads the same both ways. The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.
 * 
 * Find the largest palindrome made from the product of two 3-digit numbers.
 */

public class Euler004 {
    public static void main(String[] args){
        long startTime = System.currentTimeMillis();

        int largestPalindrome = 0;
        // Generate all products of 3-digit numbers.
        for(int i = 100; i < 999; i++){
            for(int j = i; j < 999; j++){
                // Compute reverse of product
                int trueProduct = i * j, product = trueProduct, reversedProduct = 0;

                while(product != 0){
                    reversedProduct = reversedProduct * 10 + product % 10;
                    product /= 10;
                }

                // Reassign if larger
                if(reversedProduct == trueProduct && trueProduct > largestPalindrome){
                    largestPalindrome = trueProduct;
                }
            }
        }
        System.out.println("[" + (System.currentTimeMillis() - startTime) + "ms] Solution #4: " + largestPalindrome);
        
    }
}

File: Python/Euler004.py
import time

start_time = time.time()

def is_palindrome(num):
    return str(num) == str(num)[::-1]

largest = 0
for x in range(1, 1000):
    for y in range(1, 1000):
        if(is_palindrome(x * y)):
            largest = max(largest, x * y)

print("[" + str(int(1000 * (time.time() - start_time))) + "ms] Solution #4: " + str(largest))